lib/hls/hls_classes.js

```diff
@@ -202,24 +203,41 @@ shaka.hls.Segment = class {
   /**
    * Creates an HLS segment object.
    *
-   * @param {string} absoluteUri An absolute URI.
+   * @param {string} absoluteMediaPlaylistUri An absolute Media Playlist URI.
+   * @param {string} verbatimSegmentUri verbatim segment URI.
    * @param {!Array.<shaka.hls.Tag>} tags
    * @param {!Array.<shaka.hls.Tag>=} partialSegments
    */
-  constructor(absoluteUri, tags, partialSegments=[]) {
+  constructor(absoluteMediaPlaylistUri, verbatimSegmentUri, tags,
+      partialSegments=[]) {
     /** @const {!Array.<shaka.hls.Tag>} */
     this.tags = tags;
 
-    /**
-     * An absolute URI.
-     *
-     * @const {string}
-     */
-    this.absoluteUri = absoluteUri;
+    /** @type {?string} */
+    this.absoluteUri_ = null;
+
+    /** @const {string} */
+    this.absoluteMediaPlaylistUri_ = absoluteMediaPlaylistUri;
+
+    /** @const {string} */
+    this.verbatimSegmentUri_ = verbatimSegmentUri;
 
     /** @type {!Array.<shaka.hls.Tag>} */
     this.partialSegments = partialSegments;
   }
+
+  /**
+   * Returns an absolute URI.
+   *
+   * @return {string}
+   */
+  get absoluteUri() {
+    if (!this.absoluteUri_) {
```

Comment:
Say, I just now realized this about the last PR, reading this one...
The previous changes mean that the `getUris` function that `shaka.hls.HlsParser.createSegmentReference_` creates now captures the `shaka.hls.Segment` object that was passed in, right? I don't think we were capturing those before, though I might be wrong.
Anyway, that means that those intermediate objects are being held in memory now, and thus will increase the memory footprint of each loaded segment of a parsed HLS manifest.

So maybe this should null out `this.absoluteMediaPlaylistUri_` and `this.verbatimSegmentUri_` after they are used here, since they aren't useful anymore.

Reply:
Done!